src/algebra/big_operators/basic.lean

```diff
@@ -1853,3 +1852,31 @@ begin
     simp only [his, finset.sum_insert, not_false_iff],
     exact (int.nat_abs_add_le _ _).trans (add_le_add le_rfl IH) }
 end
+
+/-! ### `additive`, `multiplicative` -/
+
+namespace finset
+open additive multiplicative
+
+section comm_monoid
+variables [comm_monoid β]
+
+@[simp] lemma of_mul_prod (s : finset α) (f : α → β) :
+  of_mul (∏ i in s, f i) = ∑ i in s, of_mul (f i) := rfl
+
+@[simp] lemma to_mul_sum (s : finset α) (f : α → additive β) :
+  to_mul (∑ i in s, f i) = ∏ i in s, to_mul (f i) := rfl
+
+end comm_monoid
+
+section add_comm_monoid
+variables [add_comm_monoid β]
+
+@[simp] lemma of_add_sum (s : finset α) (f : α → β) :
+  of_add (∑ i in s, f i) = ∏ i in s, of_add (f i) := rfl
+
+@[simp] lemma to_add_prod (s : finset α) (f : α → β) :
+  to_add (∏ i in s, f i) = ∑ i in s, to_add (f i) := rfl
+
+end add_comm_monoid
+end finset
```

Comment:
I'd argue these don't belong in the `finset` namespace, to match things like `add_monoid_hom.coe_sum` etc.

Can you add similar lemmas for `list` and `multiset` too, with names `to_add_multiset_sum` etc?